FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,57 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+
+			switch result {
+			case let .success((data, response)):
+				completion(FeedImageMapper.map(data, from: response))
+			case .failure(_):
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+}
+
+private final class FeedImageMapper {
+	private struct Root: Decodable {
+		let items: [Item]
+
+		var imageFeed: [FeedImage] {
+			return items.map { $0.item }
+		}
+	}
+
+	private struct Item: Decodable {
+		let id: UUID
+		let description: String?
+		let location: String?
+		let imageURL: URL
+
+		enum CodingKeys: String, CodingKey {
+			case id = "image_id"
+			case description = "image_desc"
+			case location = "image_loc"
+			case imageURL = "image_url"
+		}
+
+		var item: FeedImage {
+			return FeedImage(
+				id: id,
+				description: description,
+				location: location,
+				url: imageURL)
+		}
+	}
+
+	private static var statusCode_200: Int { return 200 }
+
+	internal static func map(_ data: Data, from response: HTTPURLResponse) -> RemoteFeedLoader.Result {
```

Comment:
Please remove `internal` to follow the coding convention of this codebase.

```suggestion
	static func map(_ data: Data, from response: HTTPURLResponse) -> RemoteFeedLoader.Result {
```

Reply:
You mean because it's inside a `private class`? The Challenge's Instructions says "Make careful and proper use of access control, marking as private or internal any implementation details that aren’t referenced from other external components." and in the EssentialFeed app it was also marked as internal, although it's inside a `internal final class FeedItemsMapper`.

Reply:
`internal` is a default access level. It is ok to explicitly declare it. On the other hand it is more important to be consistent across the codebase. Therefore if some class or method is explicitly marked as `internal` then all other should be marked as well. For example `Root.items` should be marked as well as `Root.feed`. This project doesn't use explicit `internals` so it should be either added everywhere or removed here.